subsys/debug/thread_analyzer.c

```diff
@@ -133,6 +138,7 @@ void thread_analyzer_auto(void)
 {
 	for (;;) {
 		thread_analyzer_print();
+		k_thread_runtime_stats_all_clear();
```

Comment:
can you explain why you have to clear the counters on every iteration of the thread analyzer? Wouldn't this only show stats since last print?

Reply:
That's correct. I find measuring stats over a well defined interval - usually the reporting interval - to be more useful than measuring from boot time, particularly when it's an action responding to user input (eg from the Zephyr shell, a thread started by other hardware input etc). Such a thread may consume a substantial amount of cpu time while it's executing, but a statistic measured since boot would only report it as using a small percentage.

For example, I had a put together a tool to measure network stack jitter, which I would configure and start from a shell command.  At the same time I wanted to gauge cpu usage of threads in the network stack. 
If I started the test after some time of the device being on, all threads would report <1% usage, when in reality they had used closer to 20% during the reporting intervals.